### ISSUE TITLE
TimberArchives 'monthly-nested' returns "too nested" (wrong) Tree

### ISSUE THREAD

**Issue by eightyfive** [jarednova/timber#812]
Following the example given in object reference (`$type = 'monthly-nested'` is default):
http://jarednova.github.io/timber/#timberarchives

I get this kind of Tree back:
```
2016
    2016
        Dec
        Oct
        ...
    2015
        Nov
        Mar
        ...
    2014
    ...

2015
    2016
        Dec
        Oct
        ...
    2015
        Nov
        Mar
        ...
    2014
    ...
2014
...
```

When I would expect (as stated in Object reference documentation):
```
2016
    Dec
    Oct
    ...
2015
    Nov
    Mar
    ...
 2014
    ...
...
```
I believe I found the root of the problem and the fix seems like an easy one:
https://github.com/jarednova/timber/blob/master/lib/timber-archives.php#L243
```php
		} elseif ( 'monthly-nested' == $type ) {
			$years = $this->get_items_yearly($args, $last_changed, $join, $where, $order, $limit);
			foreach ( $years as &$year ) {
				$args = array('show_year' => false);
				$year['children'] = $this->get_items_monthly($args, $last_changed, $join, $where, $order, $limit);
			}
			$output = $years;
		} elseif ( 'daily' == $type ) {
			...
```
`get_items_monthly(...)` _already_ returns nested items because the last parameter of the function is not passed here (and defaults to `$nested = true`).

So I would get rid of the `$years` loop and simply return the result of `get_items_monthly(..., $nested = true)`:
```php
		} elseif ( 'monthly-nested' == $type ) {
			// $years = $this->get_items_yearly($args, $last_changed, $join, $where, $order, $limit);
			// foreach ( $years as &$year ) {
			// 	$args = array('show_year' => false);
			// 	$year['children'] = $this->get_items_monthly($args, $last_changed, $join, $where, $order, $limit);
			// }
			// $output = $years;
			$output = $this->get_items_monthly($args, $last_changed, $join, $where, $order, $limit);
		} elseif ( 'daily' == $type ) {
			...
```

Advise if you want me to make pull request or if I am mistaken.

**Comment by jarednova** [comment]
Hi @eightyfive your explanation and reasoning makes sense to me. Please submit a PR and test and I'd love to merge. Thanks for the find!

**Comment by eightyfive** [comment]
Sorry a bit late: https://github.com/jarednova/timber/pull/827

**Comment by jarednova** [comment]
Closed by #827
